### PR TITLE
Proposal changes

### PR DESCRIPTION
**Pull request by MilGard91**
### Project Abstract
This is a update for previous grant [PR](https://github.com/w3f/Grants-Program/pull/255).

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [x] A BTC or Ethereum (DAI/USDT) address for the payment of the milestones is provided inside the application.
- [x] I have read and acknowledged the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).

### CODE REVIEW

**Comment by alxs** [comment]
Thanks @MilGard91. Could you please also adjust the price to account for the missing functionality and sign the CLA?

**Comment by MilGard91** [comment]
Hi @alxs, price is adjusted and I signed the CLA.

File: applications/substrate-identity-directory.md [hunk 1]
```diff
@@ -87,8 +84,7 @@ We have experience in several private and open source projects aswell as project
 | 2. | Design an individual identity page | Create design mockups in Figma. |
 | 3. | Implement list page | Build UI components in VueJS used in the identity list page. | 
 | 4. | Implement individual identity page | Build UI components in VueJS used in the individual identity page. 
-| 5. | Implement query logic for identity data | Implement query logic which will be used to retrieve identity data. |  
-| 6. | Query identity activities | Implement query logic which will be used to retrieve activities indicative of a member’s reputation. |
```
Comment (alxs) [inline applications/substrate-identity-directory.md:91]: Could you move this to a different milestone instead, together with the treasury and governance modules? Feel free to distribute the price across the two milestones as you think fit.

**Review by alxs — CHANGES_REQUESTED** [review]

File: applications/substrate-identity-directory.md (hunk 1 above)
Comment (Jakic007) [reply applications/substrate-identity-directory.md:91]: I have created Milestone 3 out of that deliverable. @alxs

Comment (alxs) [reply applications/substrate-identity-directory.md:91]: To clarify, I meant you may split the price of this milestone (which previously contained all deliverables) between this and the new milestone.

File: applications/substrate-identity-directory.md [hunk 2]
```diff
@@ -103,9 +97,23 @@ We have experience in several private and open source projects aswell as project
...
  
 
+### Milestone 3 Index data and query
+* **Estimated Duration:** 3 weeks
+* **FTE:**  2
+* **Costs:** 8166 DAI
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 1. | Index data from nodes | Setup [substrate-archive](https://github.com/paritytech/substrate-archive) with postgres for Polkadot and Kusama chains to store required data |
```
Comment (alxs) [inline applications/substrate-identity-directory.md:107]: We also only fund deliverables that may be useful to others in the ecosystem, i.e. we can't support you in integrating existing software.

File: applications/substrate-identity-directory.md [hunk 3]
```diff
@@ -103,9 +97,23 @@ We have experience in several private and open source projects aswell as project
...
+| 2. | Backend development | Provide endpoints for data queries related to governance and treasury activities |
+| 3. | Fetch data on frontend | Consume provided endpoints and display data on frontend |
+
+Mockup:  
+![Identity activities](https://i.imgur.com/h9wQwXY.png)
+
 ## Future Plans
 
-After finishing milestones 1 and 2 users can utilize the basic application which supports querying by address, index or identity fields. Application has a list page and a single page view with a basic info column fully functional. Users will be able to set their own node endpoint.
+After finishing milestones 1 and 2 users can utilize the basic application which supports querying by address. Application has a list page and a single page view with a basic info column fully functional. Users will be able to set their own node endpoint.
```
Comment (alxs) [inline applications/substrate-identity-directory.md:116]: Why did you remove this? It seems like a very useful feature to me.

Comment (alxs) [inline applications/substrate-identity-directory.md:116]: Why did you remove this? Seems like a very useful feature to me.

**Review by alxs — CHANGES_REQUESTED** [review]

File: applications/substrate-identity-directory.md (hunk 3 above)
Comment (Jakic007) [reply applications/substrate-identity-directory.md]: Accidentally. I have reverted that now.

File: applications/substrate-identity-directory.md (hunk 2 above)
Comment (Jakic007) [reply applications/substrate-identity-directory.md:107]: I am not sure if I understand. This backend infrastructure is a requirement for governance & treasury data to be retrieved.

**Comment by Jakic007** [comment]
@alxs I have just contacted Kuba to schedule another meeting, so we can solve this faster.

File: applications/substrate-identity-directory.md [hunk 4]
```diff
@@ -95,18 +95,24 @@ Mockups:
 | ------------- | ------------- | ------------- |
 | 1. | Implement logic for sending tokens | Implement logic for sending tokens; retrieve balance, parse inputs, display transaction fee, create the transfer transaction. |
 | 2. | Make the web service work offline | Web service can be used in offline mode; the user can specify a local node to which will the service connect. |
- 
 
 ### Milestone 3 Index data and query
+#### Requirements for the milestone:
+* https://github.com/paritytech/substrate-archive/issues/97
+* https://github.com/paritytech/substrate-archive/issues/111
+* https://github.com/paritytech/substrate-archive/issues/132
```
Comment (alxs) [inline applications/substrate-identity-directory.md:103]: These are different PRs from the one you mentioned in your previous email, could you please explain? And do you care to elaborate on how this is blocking you exactly? Polkascan for example have implemented pretty much this exact functionality https://polkascan.io/kusama/account/EGVQCe73TpFyAZx5uKfE1222XfkT3BSKozjgcqzLBnc5eYo

Comment (Jakic007) [reply applications/substrate-identity-directory.md:104]: I have added the missing PR to the list. To be able to retrieve and decode extrinsics from older blocks, this PR and issues must be merged and solved. Those three issues on substrate-archive are related to type-decoding.
Also, this is the response we got from the official discord server:
"At what blocks are you not getting any traces? State tracing is only supported on runtime versions v8.27 and above, so if you are on a block height below that runtime version then you will not get any traces. For runtime v0.9.XX a release is being cut to include support for those runtime versions today or tomorrow."
That is still not live in the latest version.
As agreed on the call, let's proceed with the payment for the first two milestones as they are delivered and not connected to the current issue. Thank you for your input about polkascan, we will research milestone 3 on their method for retrieving the missing data.

**Comment by Jakic007** [comment]
> I think the identity activities are the most important and useful part of the project, so I would suggest to split the price between M1 and M3 60:40, rather than the 92:8 split you are proposing. As far as I can see, the archive issues currently holding you back are already being worked on?

I agree. I have updated the cost ratio between M1 and M3

**Comment by semuelle** [comment]
You did not update the cost ratio, you just raised the total price. What makes you think we would support raising the price by more than a third without any extra deliverables?

**Comment by edisinovcic** [comment]
> You did not update the cost ratio, you just raised the total price. What makes you think we would support raising the price by more than a third without any extra deliverables?

You propose that we reduce the price of the first and second milestone vs third in 60:40 so that first 2 milestones are worth 20700$ and the 3rd milestone is worth 13800$?

**Comment by semuelle** [comment]
> You propose that we reduce the price of the first and second milestone vs third in 60:40 so that first 2 milestones are worth 20700$ and the 3rd milestone is worth 13800$?

Since M2 is unchanged, I would leave its price as is, but shift more of the payment from M1 to M3.

**Comment by edisinovcic** [comment]
> > You propose that we reduce the price of the first and second milestone vs third in 60:40 so that first 2 milestones are worth 20700$ and the 3rd milestone is worth 13800$?
> 
> Since M2 is unchanged, I would leave its price as is, but shift more of the payment from M1 to M3.

It would be amazing if you can write numbers down that you think are good for each milestone so we can fix this faster.
Milestone 1: xxx$
Milestone 2: xxx$
Milestone 3: xxx$

Tnx!

**Comment by Jakic007** [comment]
> > You propose that we reduce the price of the first and second milestone vs third in 60:40 so that first 2 milestones are worth 20700$ and the 3rd milestone is worth 13800$?
> 
> Since M2 is unchanged, I would leave its price as is, but shift more of the payment from M1 to M3.

Updated.

**Review by semuelle — APPROVED** [review]

**Review by alxs — APPROVED** [review]
I'm happy to approve this, however I don't know how likely it is the functionality discussed above will be added to subtrate-archive any time soon, so I encourage you to consider alternative solutions.

**Review by mmagician — APPROVED** [review]
